### PR TITLE
Add sorting and pagination to batch show page

### LINKED ISSUE #70: Pagination - Batch Detail
### Description
Batches with over 20 items use pagination

### Done Looks Like
- Batch with over 20 items has 20 items on the first page, and the next 20 items on page 2, etc.

### LINKED ISSUE #71: Sorting  - Batch Detail
### Description
Users can sort batch items in the batch detail page. They can sort by status and id within batch (row number or filename).

### Done Looks Like
User can sort by:
- status
- id within batch (row number or filename).

### LINKED ISSUE #72: Sorting - Batch Index View
### Description
Users can sort batch items in the batch object page. They can sort by status and id within batch (row number or filename).

### Done Looks Like
User can sort by:
- status
- id within batch (row number or filename).

### LINKED ISSUE #73: Pagination - Batch Index View
### Description
When there are over  20 batch objects viewable by a user use pagination

### Done Looks Like
- List of batch objects with over 20 items has 20 items on the first page, and the next 20 items on page 2, etc.

### PR DESCRIPTION
**Pull request by bkeese**
Fixes #70 
Fixes #71 
Fixes #72 
Fixes #73

### CODE REVIEW

File: app/views/hyrax/batch_ingest/batches/show.html.erb [hunk 1]
```diff
@@ -17,17 +27,18 @@
...
+        <th scope='col'><%= sort_link_to('ID Within Batch', 'id_within_batch') %></th>
+        <th scope='col'><%= sort_link_to('Source Location', 'source_location') %></th>
+        <th scope='col'><%= sort_link_to('Repository ID', 'object_id') %></th>
+        <th scope='col'><%= sort_link_to('Error', 'error') %></th>
       </tr>
     </thead>
     <tbody>
-      <% @presenter.batch_item_presenters.each_with_index do |batch_item_presenter, index| %>
-        <%= render 'batch_item', presenter: batch_item_presenter, index: index %>
+      <% @batch_items.each_with_index do |batch_item, index| %>
```
Comment (cjcolvar) [inline app/views/hyrax/batch_ingest/batches/show.html.erb]: Same question here about using the existing presenters vs intiatiating them.

File: app/views/hyrax/batch_ingest/batches/_batch_item.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<tr class="batch-item <%= presenter.status_css_class %>" data-batch-item-id="<%= presenter.id %>">
```
[the remark this answers is not in the document]
Comment (bkeese) [reply app/views/hyrax/batch_ingest/batches/_batch_item.html.erb]: Yes. I think this was a relic that I think should have been removed before. It is applied within the cell now, not the row.

File: app/views/hyrax/batch_ingest/batches/index.html.erb [hunk 3]
```diff
@@ -6,19 +16,20 @@
...
+        <th scope='col'><%= sort_link_to('Date', 'created_at') %></th>
+        <th scope='col'><%= sort_link_to('Items', 'batch_item_count') %></th>
+        <th scope='col'><%= sort_link_to('Submitter email address', 'submitter_email') %></th>
+        <th scope='col'><%= sort_link_to('Source Location', 'source_location') %></th>
+        <th scope='col'><%= sort_link_to('Admin Set', 'admin_set_title') %></th>
       </thead>
       <tbody>
-        <% @presenters.each do |presenter| %>
-          <%= render 'batch', presenter: presenter %>
+        <% @batches.each do |batch| %>
```
[the remark this answers is not in the document]
Comment (bkeese) [reply app/views/hyrax/batch_ingest/batches/index.html.erb]: Yes. In order for pagination to work, the iteration needs to happen on the model.

File: app/views/hyrax/batch_ingest/batches/show.html.erb (hunk 1 above)
Comment (bkeese) [reply app/views/hyrax/batch_ingest/batches/show.html.erb]: Same answer :).

**Pull request merged**